src/features/account/hooks.ts

```diff
@@ -45,33 +46,79 @@ export function useLockedBalance(address?: Address) {
   };
 }
 
-// Note, this retrieves the address' info from the Accounts contract
-// It has nothing to do with wallets or backend services
-export function useAccountDetails(address?: Address) {
+export function useVoteSigner(address?: Address) {
   const {
-    data: isRegistered,
+    data: voteSigner,
     isError,
     isLoading,
     error,
     refetch,
   } = useReadContract({
+    address: Addresses.Accounts,
+    abi: accountsABI,
+    functionName: 'voteSignerToAccount',
+    args: [address || ZERO_ADDRESS],
+    query: { enabled: !!address },
+  });
+
+  useToastError(error, 'Error fetching vote signer');
+
+  return {
+    voteSigner,
+    isError,
+    isLoading,
+    refetch,
+  };
+}
+
+// Note, this retrieves the address' info from the Accounts contract
+// It has nothing to do with wallets or backend services
+export function useAccountDetails(address?: Address, addressOrVoteSigner?: Address) {
+  const isAccountResult = useReadContract({
     address: Addresses.Accounts,
     abi: accountsABI,
     functionName: 'isAccount',
     args: [address || ZERO_ADDRESS],
     query: { enabled: !!address },
   });
 
+  const isValidatorResult = useReadContract({
+    address: Addresses.Validators,
+    abi: validatorsABI,
+    functionName: 'isValidator',
+    args: [addressOrVoteSigner || ZERO_ADDRESS],
+    query: { enabled: !!addressOrVoteSigner },
+  });
+
+  const isValidatorGroupResult = useReadContract({
+    address: Addresses.Validators,
+    abi: validatorsABI,
+    functionName: 'isValidatorGroup',
+    args: [addressOrVoteSigner || ZERO_ADDRESS],
+    query: { enabled: !!addressOrVoteSigner },
+  });
+
   // Note, more reads can be added here if more info is needed, such
   // as name, metadataUrl, walletAddress, voteSignerToAccount, etc.
 
-  useToastError(error, 'Error fetching account registration status');
+  useToastError(
+    isAccountResult.error || isValidatorResult.error || isValidatorGroupResult.error,
+    'Error fetching account details',
+  );
 
   return {
-    isRegistered,
-    isError,
-    isLoading,
-    refetch,
+    isRegistered: isAccountResult.data,
+    isValidator: isValidatorResult.data,
+    isValidatorGroup: isValidatorGroupResult.data,
+    isError: isAccountResult.isError || isValidatorResult.isError || isValidatorGroupResult.isError,
+    isLoading:
+      isAccountResult.isLoading || isValidatorResult.isLoading || isValidatorGroupResult.isLoading,
+    refetch: () =>
+      Promise.all([
+        isAccountResult.refetch,
```

Comment:
dont these need to be called?

Reply:
that's correct, they need to be recalled and probably forward the options
```ts
    refetch: (options) =>
      Promise.all([
        isAccountResult.refetch(options),
```